esp-hal/src/dma/buffers.rs

```diff
@@ -1562,3 +1562,391 @@ impl DerefMut for DmaLoopBuf {
         self.buffer
     }
 }
+
+/// A custom DMA TX buffer that lets us unsafely fiddle with the descriptors.
+///
+/// Beware the dragons.
+///
+/// This type is only meant to fill out descriptors once, then send them off to the DMA. This
+/// isn't meant to hold onto the descriptors or otherwise provide a reuseable API to work with
+/// them. In fact this should just be a way to produce a `Preparation` but we have no way to do
+/// that in the current API.
+///
+/// This type likely never should be visible outside of esp-hal.
+pub(crate) struct UnsafeTxBuffer<'a> {
+    descriptors: DescriptorSet<'a>,
+    #[cfg(psram_dma)]
+    in_psram: bool,
+}
+
+impl<'a> UnsafeTxBuffer<'a> {
+    /// Creates a new DMA bufer.
+    ///
+    /// `block_size` is the requirement imposed by the peripheral that receives the data. It
+    /// ensures that the DMA will not try to copy a partial block, which would cause the RX DMA
```

Comment:
RX?

Reply:
Yeah, RX. You try to push 8 bytes into AES using the Tx buffer, then the operation never finishes, and the RX DMA never finishes, so there is some constraint on the TX DMA channel.